sql/handler.cc

```diff
@@ -1942,6 +1944,8 @@ int ha_commit_trans(THD *thd, bool all)
   if (!is_real_trans)
   {
     error= commit_one_phase_2(thd, all, trans, is_real_trans);
+    if (error)
+      goto err;
```

Comment:
why is this? this isn't a transaction commit at all, `is_real_trans` is false

Reply:
Indeed. It does not look to relate to the current fixes.
Probably I aimed at uniformity with the above `if (error) goto err`.
Although it makes sense in this context to prefer the new `goto err`  over the current `goto done` I would not rush with the change.

Let me drop the hunk.